File: server/responses/week_admin_responses.py
import sys
import pathlib

sys.path.append(str(pathlib.Path(__file__).parent.parent))
from models import goal_models, weekly_goal_models, week_models


class CombinedGoal(goal_models.GoalInDB, weekly_goal_models.WeeklyGoalInDB):
    pass


class WeekData(week_models.WeekInDB):
    total_set: int = 0
    total_achieved: int = 0

File: client/src/types/week_admin_responses.ts
import { GoalInDB } from "./goal"
import { WeekInDB } from "./week"
import { WeeklyGoalInDB } from "./weekly_goal"

export type CombinedGoal = GoalInDB & WeeklyGoalInDB

export type WeekData = WeekInDB & {
    total_set: number
    total_achieved: number
}

